src/main/java/com/shyb/evcharging/user/application/UserService.java

```diff
@@ -0,0 +1,94 @@
+package com.shyb.evcharging.user.application;
+
+import com.shyb.evcharging.user.dto.EmailDuplicateCheckRequestDto;
+import com.shyb.evcharging.user.dto.UserModifyRequestDto;
+import com.shyb.evcharging.user.dto.UserRequestDto;
+import com.shyb.evcharging.user.dto.UserResponseDto;
+import com.shyb.evcharging.user.exception.EmailDuplicateException;
+import com.shyb.evcharging.user.exception.PasswordMisMatchException;
+import com.shyb.evcharging.user.exception.UserNotFoundException;
+import com.shyb.evcharging.user.repository.UserRepository;
+import com.shyb.evcharging.user.repository.UserSearchCond;
+import com.shyb.evcharging.utils.CustomStringUtils;
+import com.shyb.evcharging.utils.PasswordEncoder;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    /**
+     * 사용자를 등록한 후 리턴합니다.
+     *
+     * @param user 생성 요청 사용자 정보
+     * @eturn 등록된 사용자
+     */
+    public UserResponseDto save(UserRequestDto user) {
+        boolean sameString = CustomStringUtils.isSameString(user.getPassword(), user.getConfirmPassword());
+        if (!sameString) {
+            throw new PasswordMisMatchException();
+        }
+        user.passwordEncrypt();
+
+        return userRepository.save(user.toEntity());
```

Comment:
DTO의 불변이 해당 메소드 호출로 인해 깨졌고, DTO와 PasswordEncoder간의 결합이 생겼네요.

크게 중요한 부분은 아니지만 아래와 같이 정의하면 불변을 유지할 수 있지 않을까요?

```suggestion
        String encryptedPassword = ~~
        return userRepository.save(user.toEntity(encryptedPassword));
```